File: Data Structures & Algorithms/is-anagram/submission-1.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // O(m + n) - TC
        // O(26) - SC
        if (s.length() != t.length()) {
            return false;
        }

        int[] count = new int[26]; 
        for (int i = 0; i < s.length(); i++) {
            char c1 = s.charAt(i);
            char c2 = t.charAt(i);
            count[c1 - 'a'] += 1;
            count[c2 - 'a'] -= 1;
        }
        
        for (int val : count) {
            if (val != 0) {
                return false;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/task-scheduling/submission-1.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        count = {}
        for task in tasks: 
                count[task] = count.get(task,0) + 1

        maxHeap = []
        for cnt in count.values():
            if cnt > 0:
                heapq.heappush(maxHeap, -cnt)
        
        time = 0
        q = deque()
        while maxHeap or q:
            time += 1

            if maxHeap:
                cnt = abs(heapq.heappop(maxHeap)) - 1;
                if cnt > 0:
                    q.append([cnt, time + n])

            if q and q[0][1] == time: 
                heapq.heappush(maxHeap, -q.popleft()[0])


        return time